### PR TITLE
Analytics hub survey banner

### PR DESCRIPTION
**Pull request by atorresveiga**
Closes: # https://github.com/woocommerce/woomobile-private/issues/264

### Why
A feedback banner on the Analytics Hub screen will help us receive merchant feedback on the feature. With this feedback, we will be able to understand merchant needs better and plan for the next iteration of the feature.

### Description
This PR adds a feedback banner to the Analytics Hub Screen. The banner will be hidden behind a feature flag until the code reaches production. The send / dismiss logic was moved to the view model, making it accessible to unit tests. The PR also adds a reusable FeatureBanner view that could be used to add feature banners on other screens.

### Testing instructions
TC1
1. On MyStore Screen, press the `See more` button in the stats view
2. Check that the feedback request  banner is shown
3. Tap on the dismiss button (the X button on the top right)
4. Check that the request for feedback banner is hidden
5. Go back to the MyStore Screen
6. Re-open the Analytics Hub Screen by pressing the `See more` button in the stats view
7. Check that the feedback request banner is hidden

Using Flipper or another tool, remove the analytics hub status from the preferences

Screenshot 2023-03-08 at 00 38 10

TC2
1. On MyStore Screen, press the `See more` button in the stats view
2. Check that the feedback request  banner is shown
3. Tap on the SHARE FEEDBACK button
4. Check that the app navigates to the Analytics Hub survey
5. Go back to the Analytics Hub Screen
6. Check that the request for feedback banner is hidden
7. Go back to the MyStore Screen
8. Re-open the Analytics Hub Screen by pressing the `See more` button in the stats view
9. Check that the feedback request banner is hidden

### Images/gif

https://user-images.githubusercontent.com/18119390/223639694-b2012363-d318-4e04-8698-14e70cf22331.mp4

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/feedback/FeedbackBanner.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    fun setMessage(@StringRes message: Int) {
+        binding.message.setText(message)
+    }
+
+    fun setTitle(@StringRes title: Int) {
+        binding.title.setText(title)
+    }
+}
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/feedback/FeedbackBanner.kt:35]: IPP uses a similar feedback view. I only decided to go with a fresh copy to avoid stepping on our toes and because its view was tight to their scope by name and package name. In the future, if they keep the same design, maybe they should use this one, which is already in the feedback package.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/feedback/FeedbackRepository.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import javax.inject.Inject
+
+class FeedbackRepository @Inject constructor() {
+    fun getFeatureFeedback(feature: Feature): FeedbackState {
+        return FeedbackPrefs.getFeatureFeedbackSettings(feature)?.feedbackState ?: FeedbackState.UNANSWERED
+    }
+    fun saveFeatureFeedback(feature: Feature, feedbackState: FeedbackState) {
+        FeedbackPrefs.setFeatureFeedbackSettings(FeatureFeedbackSettings(feature, feedbackState))
+    }
+}
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/feedback/FeedbackRepository.kt:16]: The purpose of this class is to manage feedback data. I think the ViewModel shouldn't have any clue about the data source we are using to get/save the feedback data (in this case, Shared Preferences)

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/AnalyticsHubViewModel.kt [hunk 3]
```diff
@@ -97,6 +102,18 @@ class AnalyticsHubViewModel @Inject constructor(
...
         observeRevenueChanges()
         observeRangeSelectionChanges()
+        if (FeatureFlag.ANALYTICS_HUB_FEEDBACK_BANNER.isEnabled()) {
+            shouldAskForFeedback()
+        }
+    }
+
+    private fun shouldAskForFeedback() {
+        viewModelScope.launch {
+            val feedbackStatus = feedbackRepository.getFeatureFeedback(FeatureFeedbackSettings.Feature.ANALYTICS_HUB)
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/AnalyticsHubViewModel.kt:112]: A nice improvement could be to _`observe`_ the feedback information. As we are already using `DataStore` migrating the FeedbackPrefs to `DataStore` could be a logical next step. The migration could be of moderate effort, though. That is why I believed this migration should be tackled in a different PR.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/AnalyticsHubFragment.kt [hunk 4]
```diff
@@ -132,4 +136,19 @@ class AnalyticsHubFragment :
...
         }
     }
+
+    private fun displayFeedbackBanner(isVisible: Boolean) {
+        binding.analyticsHubFeedbackBanner.isVisible = isVisible
+        if (!isVisible) return
+        binding.analyticsHubFeedbackBanner.run {
+            onSendFeedbackListener = { viewModel.onSendFeedbackClicked() }
+            onDismissClickListener = { viewModel.onDismissBannerClicked() }
+        }
```
Comment (ThomazFB) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/AnalyticsHubFragment.kt:146]: A question: Would it make sense if we just bind the action listeners just once when the Fragment is created and leave this function only to adjust the banner visibility? 

Or maybe you did like this because you want to avoid unnecessary binding when the banner no longer gets displayed?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/feedback/FeedbackRepository.kt (hunk 2 above)
Comment (ThomazFB) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/feedback/FeedbackRepository.kt:16]: Great solution, and makes it easier to test too! 👏

**Review by ThomazFB — APPROVED** [review]
Tested and looks good! Great work, with nice code improvements! Thanks for tackling it 😄

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/AnalyticsHubFragment.kt (hunk 4 above)
Comment (atorresveiga) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/hub/AnalyticsHubFragment.kt:146]: > Or maybe you did like this because you want to avoid unnecessary binding when the banner no longer gets displayed?

It was because of this ☝️. In the planned lifecycle of the feedback banner, it will show only when the status is UNANSWERED, and once hidden, it won't be shown again.  So I thought it would be better to handle the binding in this function because it should be called only once with `isVisible=true`, in which case it will handle the bindings. For all the other cases, the function will not reach the bindings. Does that make sense?
